week4/week4.py

```diff
@@ -0,0 +1,61 @@
+# week4
+from collections import defaultdict
+
+
+class LargestSumRectangle:
+    def __init__(self, grid: list):
+        self.grid = grid
+        self.rows = len(grid)
+
+    # pre-compute row sums starting at every index ( 0(n*m) size )
+    def getRowSums(self) -> dict:
+        grid = self.grid
+        rows = self.rows
+        sums = defaultdict(int)
+
+        for i in range(rows):
+            s = 0
+            for j in range(len(grid[i])):
+                try:
+                    s += grid[i][j]
+                    sums[(i,j)] = s
+                except:
```

Comment:
Generally a good idea not to do a raw except, instead do `except SomeError` where the specific error you want to handle is `SomeError`.

Not sure in this case what that is actually. What are we excepting here?